src/code_80296A50.c

```diff
@@ -3017,60 +3006,40 @@ void place_all_item_boxes(struct ActorSpawnData *spawnData) {
     }
 }
 
-#ifdef MIPS_TO_C
-//generated by mips_to_c commit 3c3b0cede1a99430bfd3edf8d385802b94f91307
-s32 func_8029EC88(? *, ? *, ? *, ?); // extern
-extern ? gActorList;
-extern u16 gPlayers;
-
 void func_8029D9F8(void) {
-    ? sp64;
-    ? sp58;
-    ? sp50;
-    s16 temp_s0;
-    u16 temp_v0;
-    u16 *phi_s1;
+    Vec3f sp64;
+    Vec3f sp58;
+    Vec3s sp50;
+    Player *phi_s1;
+    struct Actor *actor;
     s16 phi_s0;
+    s32 i;
 
-    phi_s1 = &gPlayers;
-    phi_s0 = 0;
-    do {
-        temp_v0 = *phi_s1;
-        if (((temp_v0 & 0x4000) != 0) && ((temp_v0 & 0x100) == 0)) {
-            (&gActorList + (func_8029EC88(&sp64, &sp50, &sp58, 0x2D) * 0x70))->unk_04 = phi_s0;
-        }
-        temp_s0 = phi_s0 + 1;
-        phi_s1 += 0xDD8;
-        phi_s0 = temp_s0;
-    } while (temp_s0 != 4);
+    //phi_s0 = 0;
+    for (i = 0; i < 4; i++) {
+        phi_s1 = &gPlayers[i];
+        //temp_v0 = *phi_s1;
+        if ((phi_s1->unk_000 & 0x4000) == 0) { continue; }
+        if ((phi_s1->unk_000 & 0x100) != 0) { continue; }
+
+        phi_s0 = func_8029EC88(sp64, sp50, sp58, (s16)45);
```

Comment:
> arbitrary values from the stack
They do this a lot right?

Yea, order matters for the sp's. iirc m2c didn't place sp50 in the right spot. There was definitely some messing around plus I believe the continue statements were required here rather than `if () {// do stuff}`. Tricky stuff for sure.